src/components/proofs/EmailProof.tsx

```diff
@@ -81,12 +81,29 @@ const questionBlock = (open: boolean) =>
 
 const tooltipWrapper = classnames(display('flex'), flex('flex-1'))
 
-export default function () {
-  const [domain, setDomain] = useState('')
-  const [open, setOpen] = useState(false)
+export default function ({
+  domain: defaultDomain,
+  token: defaultToken,
+}: {
+  domain?: string
+  token?: string
+}) {
+  const { emailDomain } = useSnapshot(EmailDomainStore)
+  const [domain, setDomain] = useState(defaultDomain ?? '')
```

Comment:
So, `defaultDomain` may be empty? Btw, you're doing this operation twice, worth extracting into `const`